### PR TITLE
Add undo ability for reviewers

### LINKED ISSUE #68: Ability for moderators to "undo accept"

### PR DESCRIPTION
**Pull request by Merlin04**
This PR adds the ability to undo a review decision. It leaves logs in the channel about the decision that was being undone and who undid it to reduce likelihood of abuse. I've also refactored some components (extracted out interaction handlers into separate files and made a single system to pack data into view callback ids), updated some dependencies, and fixed some bugs. (closes #68)

### CODE REVIEW

**Comment by Merlin04** [comment]
oh, also - you'll need to update the oauth scopes to match what's in the updated manifest

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "prox2",
   "version": "1.0.0",
   "scripts": {
-    "dev": "next",
+    "dev": "env NODE_OPTIONS=--openssl-legacy-provider next",
```
Comment (anirudhb) [inline package.json:5]: Please do not include hacks specific to your machine in the PR

**Review by anirudhb — CHANGES_REQUESTED** [review]

**Comment by anirudhb** [comment]
Build failed, take a look

**Review by anirudhb — APPROVED** [review]
tested, seems to work!

**Pull request merged**